File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.ts
function twoSum(nums: number[], target: number): number[] {
    let map = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        let comp = target - nums[i];
        if (map.has(comp)) {
            return [map.get(comp) + 1, i + 1];
        }
        map.set(nums[i], i);
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int l = 0, r = nums.length - 1;
        while (l < r) {
            int sum = nums[l] + nums[r];
            if (sum > target)
                r--;
            else if (sum < target)
                l++;
            else {
                return new int[] { l+1,r+1 };
            }
        }
        return new int[] {};
    }
}